File: backend/schemas/feedback.py
"""反馈相关请求/响应模型

对应 SDD 4.5.9 反馈接口。
"""
from __future__ import annotations

from pydantic import BaseModel, Field


class FeedbackRequest(BaseModel):
    """质量反馈请求。

    :param translation_id: 关联的翻译记录 ID
    :param type: 反馈类型 accurate/inaccurate/outdated
    :param comment: 补充说明，最长 500 字
    """

    translation_id: int = Field(..., description="关联的翻译记录 ID")
    type: str = Field(..., description="反馈类型：accurate/inaccurate/outdated")
    comment: str | None = Field(None, max_length=500, description="补充说明")


class FeedbackResponse(BaseModel):
    """反馈响应数据。"""

    success: bool = True
    message: str = "反馈成功"
    feedback_id: int = 0

File: app/src/api/feedback.js
/**
 * 反馈接口
 * 对应后端：backend/api/v1/feedback.py
 * - POST /feedback 提交质量反馈
 */
import { post } from './request'

/**
 * 提交质量反馈
 * @param {object} data
 * @param {number} data.translation_id 翻译记录 ID
 * @param {string} data.type 反馈类型：accurate/inaccurate/outdated
 * @param {string} data.comment 备注（可选）
 * @returns {Promise<{success: boolean, message: string, feedback_id: number}>}
 */
export function submitFeedback(data) {
  return post('/feedback', data)
}

export default {
  submitFeedback
}
